data_steward/cdr_cleaner/manual_cleaning_rules/update_questiona_answers_not_mapped_to_omop.py

```diff
@@ -0,0 +1,167 @@
+# coding=utf-8
+"""
+Update survey Questions and Answers not mapped to OMOP concepts.
+
+There are several survey questions and answers that are not getting properly mapped from the RDR into the CDR.
+There are two sources for this error:
+    > Odysseus attempted to introduce some “short codes” that were not implemented in the RDR.
+    > There were some concepts that were invalidated and changed around.
+To address this, we implemented a fix that maps from the source_values present in the data to the “real”
+concept_ids in the OMOP vocabulary. This includes leveraging the supplemental old_map_short_codes.CSV where,
+Odyseus provided the short codes, as well as a brief review of other outstanding unmapped PPI codes.
+"""
+import logging
+
+import bq_utils
+import constants.cdr_cleaner.clean_cdr as cdr_consts
+import sandbox
+
+LOGGER = logging.getLogger(__name__)
+
+OLD_MAP_SHORT_CODES_TABLE = 'old_map_short_codes'
+
+OLD_MAP_SHORT_CODES_TABLE_FIELDS = [
+    {
+        "type": "string",
+        "name": "type",
+        "mode": "nullable"
+    },
+    {
+        "type": "string",
+        "name": "pmi_code",
+        "mode": "nullable"
+    },
+    {
+        "type": "string",
+        "name": "short_code",
+        "mode": "nullable"
+    }
+]
+
+UPDATE_QUESTIONS_MAP_QUERY = """
+    UPDATE
+        `{project}.{dataset}.observation` obs
+    SET
+        observation_concept_id=new_observation_concept_id,
+        observation_source_concept_id=new_observation_source_concept_id
+    FROM
+        (SELECT
+        DISTINCT short_pmi_code AS observation_source_value,
+        source_c.concept_id AS new_observation_source_concept_id,
+        FIRST_VALUE(standard_c.concept_id) OVER (PARTITION BY source_c.concept_id ORDER BY c_r.relationship_id DESC ) AS new_observation_concept_id
+        FROM (
+            SELECT
+            SUBSTR(pmi_code,1,50) AS short_pmi_code,
+            short_code
+            FROM
+                `{project}.{sandbox}.{old_map}`
+            WHERE
+                type='Question' )
+        LEFT JOIN
+            `{project}.{dataset}.concept` source_c
+        ON
+            (short_code=concept_code)
+        JOIN
+            `{project}.{dataset}.concept_relationship` c_r  
+        ON
+            (source_c.concept_id=c_r.concept_id_1)
+        JOIN
+            `{project}.{dataset}.concept` standard_c
+        ON
+            (standard_c.concept_id=c_r.concept_id_2)
+        WHERE
+            source_c.vocabulary_id='PPI'
+            AND c_r.relationship_id LIKE 'Maps to%'
+        ) map
+    WHERE
+        map.observation_source_value=obs.observation_source_value"""
+
+UPDATE_ANSWERS_MAP_QUERY = """
```

Comment:
This is duplicating the query above.  Suggest removing one query, renaming the other to something like 'UPDATE_MAP_QUERY' and parameterizing the type.

Reply:
there are also some changes in the fields, thats the reason i used different queries.

Reply:
You're right.  Parameterizing this would add 4-5 extra string parameters.  This is fine.